scripts/benchmarks_ci.py

```diff
@@ -221,25 +222,37 @@ def __main(args: list) -> int:
 
     # Run micro-benchmarks
     if not args.build_only:
-        for framework in args.frameworks:
-            micro_benchmarks.run(
-                BENCHMARKS_CSPROJ,
-                args.configuration,
-                framework,
-                verbose,
-                args
-            )
-            
-        dotnet.shutdown_server(verbose)
-
-        if args.upload_to_perflab_container:
-            import upload
+        upload_container = UPLOAD_CONTAINER
+        try:
+            for framework in args.frameworks:
+                micro_benchmarks.run(
+                    BENCHMARKS_CSPROJ,
+                    args.configuration,
+                    framework,
+                    verbose,
+                    args
+                )
             globpath = os.path.join(
                 get_artifacts_directory() if not args.bdn_artifacts else args.bdn_artifacts,
                 '**',
                 '*perf-lab-report.json')
+        except CalledProcessError:
+            upload_container = 'failedresults'
+            globpath = os.path.join(
+                get_artifacts_directory(), 
```

Comment:
this use of `get_artifacts_directory()` is also not correct for the lab.

Reply:
Should I replace it with get_artifacts_directory() if not args.bdn_artifacts else args.bdn_artifacts as in the above case?

Reply:
Yes, this will ensure that the failure json will get written into the correct artifacts location.